File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if(nums.empty()) return 0;

        int count = 1;
        int maxCount = 1;

        sort(nums.begin(), nums.end());

        for(int i = 1; i < nums.size(); i++) {

            // Duplicate
            if(nums[i] == nums[i-1])
                continue;

            // Consecutive
            if(nums[i] == nums[i-1] + 1)
                count++;
            else
                count = 1;

            maxCount = max(count, maxCount);
        }

        return maxCount;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution(object):
    def longestConsecutive(self, nums):
        if not nums:
            return 0

        nums.sort()

        count = 1
        maxi = 1

        for i in range(1, len(nums)):
            if nums[i] == nums[i-1] + 1:
                count += 1

            elif nums[i] == nums[i-1]:
                continue

            else:
                count = 1

            maxi = max(maxi, count)

        return maxi